qip/examples/deutch.rs

```diff
@@ -0,0 +1,73 @@
+#[cfg(feature = "macros")]
+use qip::builder::Qudit;
+#[cfg(feature = "macros")]
+use qip::builder_traits::UnitaryBuilder;
+#[cfg(feature = "macros")]
+use qip::prelude::*;
+#[cfg(feature = "macros")]
+use std::num::NonZeroUsize;
+
+#[cfg(feature = "macros")]
+fn apply_oracle<F>(
+    b: &mut dyn UnitaryBuilder,
```

Comment:
UnitaryBuilder doesn't exist anymore - please replace with LocalBuilder or a generic trait.

Reply:
Sorry, it does exist but isn't meant to be used with `dyn` - rather just as a generic trait. So switch to either generics or the `LocalBuilder` struct.